homeassistant/components/myuplink/__init__.py

```diff
@@ -96,3 +97,11 @@ def create_devices(
                     sw_version=device.firmwareCurrent,
                     serial_number=device.product_serial_number,
                 )
+
+
+async def async_remove_config_entry_device(
+    hass: HomeAssistant, config_entry: ConfigEntry, device_entry: DeviceEntry
+) -> bool:
+    """Remove myuplink config entry from a device."""
+
+    return True
```

Comment:
OK, but there is a not uncommon use case in this integration for a more permitting approach.
The myuplink API is a moving target and sensors are sometimes created and then removed in the API a couple of weeks later. If you see a bunch of stale entities you can then delete the device and it will be automatically recreated after reload of the integration or after a restart of HA. You can of course delete the entities manually but there is a heatpump model that created >600 entities where 580 later went stale. 
So I would prefer if we could leave it like this. I can see that there are numerous integrations in the code-base that uses this somewhat simplistic approach.

Reply:
We have discussed this internally, and in this case `if it can check, so it should`
So please add a check to make sure it is orphaned before returning `True`.

Reply:
Done!